src/main/java/net/goldiriath/plugin/game/loot/LootManager.java

```diff
@@ -109,38 +108,27 @@ protected void onStop() {
     @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
     public void onPlayerKillMob(EntityDeathEvent event) {
         LivingEntity mob = event.getEntity();
-        if (!(mob.getLastDamageCause() instanceof EntityDamageByEntityEvent)) {
-            return;
-        }
-        EntityDamageByEntityEvent damageEvent = (EntityDamageByEntityEvent) mob.getLastDamageCause();
-        if (!(damageEvent.getDamager() instanceof Player)) {
-            return;
-        }
-        Player player = (Player) damageEvent.getDamager();
 
-        // TODO: This does not work!
         NPC npc = plugin.msm.getBridge().getNPC(mob);
         if (npc == null) {
-            logger.warning(player.getName() + " killed mob that is not a NPC!");
             return;
         }
 
-        MobProfileTrait trait = npc.getTrait(MobProfileTrait.class);
+        HostileMobTrait trait = npc.getTrait(HostileMobTrait.class);
         if (trait == null) {
-            logger.warning(player.getName() + " killed NPC that does not have a mobspawn profile!");
+            logger.warning("An NPC died that did not have a mobspawn profile!");
             return;
         }
 
         MobSpawnProfile profile = trait.getProfile();
-
-        LootProfile loot = profile.getLoot();
+        LootProfile loot = profile.getLootProfile();
         if (loot == null) {
-            logger.warning(player.getName() + " killed NPC that does not have a loot profile!");
             return;
         }
 
-        List<ItemStack> drops = loot.drop(profile.getTier());
-        player.getInventory().addItem(drops.toArray(new ItemStack[]{}));
+        Player lastAttacker = plugin.getServer().getPlayer(trait.getLastAttacker());
+        List<ItemStack> drops = loot.drop(profile.getLootTier());
+        lastAttacker.getInventory().addItem(drops.toArray(new ItemStack[]{}));
```

Comment:
@cashenkes What do you think is appropriate here? Divide the drops randomly among all players whom have attacked the mob? What about parties? What if aiding players did not do direct damage, but did help (e.g. healers)?